# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# Filename: package/.../transcode/hotfix-filter_logo.patch
# Copyright (C) 2006 The OpenSDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- transcode-1.0.2/filter/filter_logo.c	2005-07-04 09:09:32.000000000 +0200
+++ transcode/filter/filter_logo.c	2006-10-08 00:22:48.000000000 +0200
@@ -146,6 +146,7 @@
   }
 
   if(ptr->tag & TC_FILTER_INIT) {
+    int dry_run = 0;
 
     if((vob = tc_get_vob())==NULL) return(-1);
 
@@ -181,8 +182,10 @@
 	if (optstr_get (options, "rgbswap", "") >= 0) mfd->rgbswap=!mfd->rgbswap;
 	if (optstr_get (options, "grayout", "") >= 0) mfd->grayout=!mfd->grayout;
 
-	if (optstr_get (options, "help", "") >= 0)
+	if (optstr_get (options, "help", "") >= 0) {
 	    help_optstr();
+	    dry_run = 1;
+	}
     }
 
     if (verbose > 1) {
@@ -195,6 +198,9 @@
 	printf ("       ignoredelay = %d\n", mfd->ignoredelay);
 	printf ("           rgbswap = %d\n", mfd->rgbswap);
     }
+    if (dry_run) {
+	return(0);
+    }
 
     InitializeMagick("");
 
